Automattic/WP-Job-Manager
# Job Submission Form: Validate Description and Category Fields Client Side

Add js validation checks for the Description and Category fields on the job submission form.

Fixes #1163

#### Changes proposed in this Pull Request:

* We add checks in js to validate the fields that don't work `out of the box` using HTML 5 form validation.
* If the validation fails, we use `setCustomValidity` to show a message. Note: This currently uses hard-coded english strings in the js. Any options there for localization?
* I had to do a hacky workaround to get the validation message to show for the `Description` field. Since it uses `tinymce`, the `<textarea>` is hidden, so I had to unhide it and make it 1 pixel tall... seems ok from my testing but could be a bit fragile.

#### Testing instructions:

* Create a job posting, but leave the Description field blank. You should see this message:
<img width="551" alt="Screen Shot 2019-04-30 at 3 24 24 PM" src="https://user-images.githubusercontent.com/789137/56997657-6872df00-6b5d-11e9-924c-9a6c9b66bab2.png">

* Enter something in the Description field, and save the posting. It should pass validation and show the preview.
* Enable categories in `/wp-admin/edit.php?post_type=job_listing&page=job-manager-settings#settings-job_listings`.
* Make another posting but don't select a category, you should see a message:
<img width="550" alt="Screen Shot 2019-04-30 at 3 26 31 PM" src="https://user-images.githubusercontent.com/789137/56997704-92c49c80-6b5d-11e9-9120-acbbb563eb14.png">

* Choose a category, and save the submission. It should pass validation and show the preview.


#### Proposed changelog entry for your changes: Description and Category fields are now validated before a job posting is submitted.
